From 53142302c0794006261d97af3ffcab043f502d10 Mon Sep 17 00:00:00 2001
From: Awais Belal <awais_belal@mentor.com>
Date: Fri, 15 Dec 2017 12:48:10 +0500
Subject: [PATCH 1/2] When building llvm from git or svn it embeds the svn/git
 revision into internal version string

$ /mnt/a/oe/build/tmp/work/corei7-64-bec-linux/mesa/2_17.1.5-r0/recipe-sysroot/usr/lib/llvm5.0/llvm-config-host --version
5.0.0git-9a5c333388c

We need to ignore everything after 5.0.0 which is what the cut cmd is doing

Upstream-Status: Pending
Signed-off-by: Khem Raj <raj.khem@gmail.com>
Signed-off-by: Awais Belal <awais_belal@mentor.com>
---
 configure.ac | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index 8c95007..9c92b28 100644
--- a/configure.ac
+++ b/configure.ac
@@ -995,7 +995,7 @@ strip_unwanted_llvm_flags() {
 
 llvm_set_environment_variables() {
     if test "x$LLVM_CONFIG" != xno; then
-        LLVM_VERSION=`$LLVM_CONFIG --version | egrep -o '^[[0-9.]]+'`
+        LLVM_VERSION=`$LLVM_CONFIG --version | cut -c1-5`
         LLVM_CPPFLAGS=`strip_unwanted_llvm_flags "$LLVM_CONFIG --cppflags"`
         LLVM_INCLUDEDIR=`$LLVM_CONFIG --includedir`
         LLVM_LIBDIR=`$LLVM_CONFIG --libdir`
@@ -2639,7 +2639,7 @@ detect_old_buggy_llvm() {
     dnl ourselves.
     dnl (See https://llvm.org/bugs/show_bug.cgi?id=6823)
     dnl We can't use $LLVM_VERSION because it has 'svn' stripped out,
-    LLVM_SO_NAME=LLVM-`$LLVM_CONFIG --version`
+    LLVM_SO_NAME=LLVM-`$LLVM_CONFIG --version|cut -c1-5`
     AS_IF([test -f "$LLVM_LIBDIR/lib$LLVM_SO_NAME.$IMP_LIB_EXT"], [llvm_have_one_so=yes])
 
     if test "x$llvm_have_one_so" = xyes; then
-- 
2.11.1

